packages/node/src/providers/state.ts

```diff
@@ -1,24 +1,24 @@
 import * as evm from '../evm';
+import { getConfigSecret } from '../config';
 import { removeKeys } from '../utils/object-utils';
-import {
-  ChainConfig,
-  ChainProvider,
-  ChainType,
-  EVMProviderState,
-  Config,
-  ProviderSettings,
-  ProviderState,
-} from '../types';
+import { ChainConfig, ChainType, EVMProviderState, Config, ProviderSettings, ProviderState } from '../types';
 import { BLOCK_COUNT_HISTORY_LIMIT, BLOCK_COUNT_IGNORE_LIMIT, BLOCK_MIN_CONFIRMATIONS } from '../constants';
 
 export function buildEVMState(
   coordinatorId: string,
   chain: ChainConfig,
-  chainProvider: ChainProvider,
+  chainProviderName: string,
   config: Config
 ): ProviderState<EVMProviderState> {
   const masterHDNode = evm.getMasterHDNode();
-  const provider = evm.buildEVMProvider(chainProvider.url, chain.id);
+  const chainProviderEnvironmentConfig = config.environment.chainProviders.find(
+    (c) => c.chainType === chain.type && c.chainId === chain.id && c.name === chainProviderName
+  );
+  if (!chainProviderEnvironmentConfig) {
+    throw new Error('Chain provider environment variable name must be defined in the provided config');
```

Comment:
Not sure if I should be throwing an error or doing something else here

Reply:
I'm fine with throwing but I think the error message should include the context we already have (chain type, ID and provider name)